psi4/driver/p4util/exceptions.py

```diff
@@ -85,6 +85,7 @@ class ConvergenceError(PsiException):
     def __init__(self, eqn_description, maxit):
         msg = "Could not converge %s in %d iterations." % (eqn_description, maxit)
         PsiException.__init__(self, msg)
+        self.maxit = maxit
```

Comment:
Maybe `self.iteration` or `self.iter`? This could be thrown under < maxit circumstances. In fact, someday there might be `MaxiterExceededSCFConvergenceError`, `OscillationDetectedSCFConvergenceError`, etc.

Reply:
`self.iteration` would be a huge help. I would also suggest `self.e_convergence` and `self.d_convergence` would be a huge future boon so that it allows users to inspect some of the circumstances of the failure and determine if it passes a "good enough" mark.

Reply:
`self.maxit` _is_ the iteration number. You're agreeing to rename the exception member data?

I think e_ and d_convergence can be extracted from the wfn. You're suggesting appending current energy and current rms density to error?

Reply:
Yes to renaming the member data.

Unless I am missing something it does not appear that you can extract these quantities from the Wavefunction. At the same time I would argue iter/e_conv/d_conv data makes up the current "convergence state" of the wavefunction and should be carried around together so either get all of these variables from the wavefunction or get none.